### DISCUSSION TITLE
/hyndsight/m4comp-response/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#283] [Announcements]
M4 Forecasting Competition: response from Spyros Makridakis 
 https://robjhyndman.com/hyndsight/m4comp-response/

**Comment by robjhyndman** [comment]
```header
Originally posted by AnscombesGimlet on 2017-11-21 15:54:20
````
Glad to hear there are some sub-monthly frequencies in this. It will be
interesting to see how far univariate methods have come since the last
competition. Particularly curious about the performance of temporal
hierarchies and bagging methods. Will you be participating professor?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-11-22 00:46:56
````
Most likely one or two of my students will participate.

**Comment by robjhyndman** [comment]
```header
Originally posted by Schiaparelli on 2017-11-21 19:25:38
````
Very relevant to include series with higher frequencies. There are
numerous use cases (forecasting second-to-second grid frequency etc.)
and a lot of public data sources. Maybe they should ask for suggestions
from the community next time?

However, from my perspective the univariate forecasting competitions are
a bit &quot;constructed&quot;. As I understand it, they don\'t include
explanatory variables? In real world applications you usually have at
least some information captured in explanatory variables at your
disposal. Are there any similar competitions testing automated forecasts
with explanatory variables?

&quot;The winner of M4 will be determined by evaluating the average accuracy
of the point forecasts provided by weighting two or three accuracy
metrics in a way that will be announced on the M4 website before January
1, 2018.&quot; - Interesting which accuracy metrics they settle on (and how
they justify the weighting scheme)

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-11-22 00:46:33
````
The tourism competition included explanatory variables. See
[https://robjhyndman.com/pub\...](https://robjhyndman.com/publications/the-tourism-forecasting-competition/ &quot;https://robjhyndman.com/publications/the-tourism-forecasting-competition/&quot;).
However, it wasn\'t open for public submission. The later public stage
of the competition did not include explanatory variables. There are lots
of kaggle competitions and some of them involve time series with
explanatory variables.

**Comment by robjhyndman** [comment]
```header
Originally posted by Mark Landry on 2017-12-01 09:22:12
````
&quot;\
Although a valid point, this would have excluded more than 95% of the
methods (including ML and those based on exponential smoothing) that
cannot produce prediction intervals, not to mention &quot;full distribution
forecasts&quot;\
&quot;

ML methods that solve for quantiles can produce such distribution
forecasts. While not all methods support quantiles, quite a lot of off
the shelf implementations do, particularly boosted trees used
pervasively in Kaggle (e.g. gbm, xgboost, lightGbm). They were used to
win exactly such a task in GEFCom2014 (wind, 1st &amp; 2nd). It is worth
mentioning that directly solving for them in this way is different than
the NNETAR prediction interval method mentioned in the follow-up.

However, I do agree that the additional complexity of generating such
distributions will discourage those that are unsure about entering.
Making it optional then adds a time-consuming task for those competing,
so I think it unlikely that many will opt to submit those intervals.
Especially if the evaluation metrics chosen are misaligned with
distribution evaluation\--the participant will have little incentive to
add code and processing time to generate those. I\'d argue that solving
directly for them will provide proper evidence that the best methodology
has been chosen for such a purpose. Those that provide them because they
come &quot;free&quot; with the method chosen, have likely not optimized for
them, and it would be difficult to conclude much with such a potential
bias.

In any case, it still looks like an exciting competition! Thanks for
your efforts in setting it up and encouraging participation.
